### DISCUSSION TITLE
Default sampler list seems inconsistent

### DISCUSSION THREAD

**Discussion by AlUlkesh** [vladmandic/sdnext#537] [General]
The current list: https://github.com/vladmandic/automatic/blob/def7a02d82f4e7456177f3dbdf697dc7d02a3214/modules/shared.py#L433

has "DPM++ SDE" twice. And I wonder how this list was chosen in the first place. I would think most people would want to use the newest samplers of each type, so perhaps

Euler a
DPM++ 2S a Karras
DPM++ 2M Karras
DPM++ SDE Karras
UniPC

What would you expect to be in the default list?

**Comment by vladmandic** [comment]
I'm open to suggestions - awaiting community feedback on what you'd like to see as a default list.
I want to keep it relatively short as if two samplers are nearly identical, i don't see value of having them in UI for average users and advanced users can always enable them in Settings.

**Reply by Yodrik** [reply]
A setting for a default sampler (maybe even per generation type) rather than changing the list itself would be preferred. I understand it would probably fall into a QOL request alongside saving default generation settings, etc..

**Comment by vdruts** [comment]
I agree with OP on list structure.

**Reply by vladmandic** [reply]
sorry, not sure i understand - agree with what?

**Comment by Tillerz** [comment]
I always use Euler A right now for everything. Anyone has any use cases where you use different ones? UniPC is the best one for realistic results, I think?

**Reply by brknsoul** [reply]
I use DPM++ 2M Karras, since it's by and far a better sampler than Euler a, needing fewer steps, see a plot I did here;
https://user-images.githubusercontent.com/1969381/239793292-e86cf315-9f5d-4db3-ab49-7215f02abce2.jpg

**Reply by MDMAchine** [reply]
If I want speed DPM++ 2M Karras. If I want quality DPM++ SDE Karras. 36 steps,  SDE you can get as low as 15 due to it doing twice the work per step. And SDE you have to accept it would be difficult or impossible to replicate exactly.

**Reply by brknsoul** [reply]
I used to use SDE. But for just messing around I use 2M. Heun seems pretty decent for low steps, too.

**Reply by AlUlkesh** [reply]
> If I want speed DPM++ 2M Karras. If I want quality DPM++ SDE Karras.

Same for me. I very seldomly use another one besides those 2.
